app/rewards/service.py

```diff
@@ -16,6 +16,25 @@ def create_unique_referral_code() -> str:
     return code
 
 
+def add_referral_code(code: str):
+    """
+    Add a new referral code to the referral codes collection
+
+    :param code: The referral code to add
+    """
+    firestore_db.collection(REFERRAL_CODES_COLLECTION).document(code).set({})
+    # TODO: add property to each referral code so that they can be traced back to original user
```

Comment:
You can add the user email here if you want. Though it was more just to guarantee no duplicates since it's already saved in the user collection.

Reply:
It's actually probably fine if we don't. We already have a way to get a user given a referral code